datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -1321,6 +1321,20 @@ impl LogicalPlan {
             | LogicalPlan::Extension(_) => None,
         }
     }
+
+    pub fn contains_outer_reference(&self) -> bool {
```

Comment:
If we are going to make this a public functions, I think it should have comments. For example:

```suggestion
    /// If this node's expressions contains any references to an outer subquery
    pub fn contains_outer_reference(&self) -> bool {
```